### PR TITLE
Reload rules configmap

### PR DESCRIPTION
**Pull request by clyang82**
fix this issue - https://github.com/observatorium/operator/issues/19

### CODE REVIEW

**Comment by clyang82** [comment]
@squat @metalmatze Please take a look if you agree with this approach to fix the reload configmap issue.

If so, I will do add some tests to be validate by `kubeval`. Thanks.

**Comment by marcolan018** [comment]
@metalmatze 
can you help to review this PR?
We know that we need to recreate the PR, but want to check whether the approach is ok before that.

File: all.jsonnet [hunk 1]
```diff
@@ -47,11 +47,13 @@ local re = t.receive(commonConfig {
...
 });
 
-
 local ru = t.rule(commonConfig {
+  local cfg = self,
   replicas: 1,
   rulesConfig: [{ name: 'test', key: 'test' }],
   alertmanagersURLs: ['alertmanager:9093'],
+  reloaderImage: 'quay.io/openshift/origin-configmap-reloader:' + cfg.reloaderVersion,
+  reloaderVersion: '4.5.0',
```
Comment (metalmatze) [inline all.jsonnet:56]: This could just be reloaderImage that includes the version?

File: jsonnet/kube-thanos/kube-thanos-rule.libsonnet [hunk 2]
```diff
@@ -133,6 +135,20 @@ function(params) {
       terminationMessagePolicy: 'FallbackToLogsOnError',
     };
 
+    local reloadC = {
```
Comment (metalmatze) [inline jsonnet/kube-thanos/kube-thanos-rule.libsonnet:138]:
```suggestion
    local reloadContainer = {
```

File: jsonnet/kube-thanos/kube-thanos-rule.libsonnet [hunk 3]
```diff
@@ -150,7 +166,8 @@ function(params) {
             labels: tr.config.commonLabels,
           },
           spec: {
-            containers: [c],
+            containers: [c] + 
+              (if std.length(tr.config.rulesConfig) > 0 then [reloadC] else []),
```
Comment (metalmatze) [inline jsonnet/kube-thanos/kube-thanos-rule.libsonnet:170]: Is there any reason not to simply always add this?

Comment (clyang82) [reply jsonnet/kube-thanos/kube-thanos-rule.libsonnet:170]: I do think that we need to add this container only if we have `rulesConfig` defined. if there is not `--rule-file` defined, we do not need to have this feature. WDYT?

File: all.jsonnet (hunk 1 above)
Comment (clyang82) [reply all.jsonnet]: of course. just notice that I did not expose the `reloaderVersion` in annotation. removed `reloaderVersion`

File: all.jsonnet [hunk 4]
```diff
@@ -47,11 +47,11 @@ local re = t.receive(commonConfig {
   hashringConfigMapName: 'hashring',
 });
 
-
 local ru = t.rule(commonConfig {
   replicas: 1,
   rulesConfig: [{ name: 'test', key: 'test' }],
   alertmanagersURLs: ['alertmanager:9093'],
+  reloaderImage: 'quay.io/openshift/origin-configmap-reloader:4.5.0',
```
Comment (simonpasquier) [inline all.jsonnet:54]: why not use the upstream image from https://github.com/jimmidyson/configmap-reload? another option would be to use [quay.io/prometheus-operator/prometheus-config-reloader](https://quay.io/repository/prometheus-operator/prometheus-config-reloader) which should provide the same features and is more robust IMHO.

Comment (clyang82) [reply all.jsonnet:54]: Good point. Thanks @simonpasquier Let me update to use upstream image.

**Review by kakkoyun — COMMENTED** [review]
This looks good. If you rebase it we can merge.

**Comment by clyang82** [comment]
@kakkoyun rebased. Thanks.

**Review by kakkoyun — APPROVED** [review]
🥇

**Pull request merged**